server/api/user_handler.py

```diff
@@ -23,35 +19,71 @@ def get(self):
+        valid_data = None
         try:
-            user_data = user_schema.load(req_data)
-        except Exception:
-            return custom_json_response({
-                "error": "Please supply a valid email and password."
-            }, 400)
+            # load method from marshmallow validates data according to schema definition
+            valid_data = user_schema.load(req_body)
+        except ValidationError as err:
+            return custom_json_response(err.messages, 400)
 
-        if User.get_user_by_email(req_data.get("email")):
+        if User.get_user_by_email(req_body['email']):
             return custom_json_response({
                 "error": "User already exists. Please supply a different email."
             }, 400)
 
-        new_user = User(**user_data)
+        new_user = User(**valid_data)
         new_user.save()
 
         access_token = create_access_token(
             identity={"id": new_user.id}, expires_delta=timedelta(days=30))
+
         refresh_token = create_refresh_token(
             identity={"id": new_user.id}, expires_delta=timedelta(days=30))
-        
-        data = {
+
+        response_data = {
             "message": "Created",
             "user": user_schema.dump(new_user)
         }
-        response = custom_json_response(data, 201)
-        
+        response = custom_json_response(response_data, 201)
+
         set_access_cookies(response, access_token)
         set_refresh_cookies(response, refresh_token)
-        
+
         return response
+
+    @jwt_required
+    def put(self):
+        req_body = request.get_json()
+        try:
+            valid_data = user_schema.load(req_body, partial=True)
+        except ValidationError as err:
+            return custom_json_response(err.messages, 400)
+
+        current_userid = get_jwt_identity()
+        user = User.get_one_user(current_userid)
+
+        geocode_result = geocoder(valid_data.get('streetAddress', ''), valid_data.get(
+            'city', ''), valid_data.get('state', ''), valid_data.get('zipcode', ''), valid_data.get('country', ''))
+
+        location = geocode_result.json()['results'][0]
+
+        latitude = location['geometry']['location']['lat']
+        longitude = location['geometry']['location']['lat']
+        formatted_address = location['formatted_address']
+
+        valid_data['latitude'] = float(latitude)
+        valid_data['longitude'] = float(longitude)
+        valid_data['formattedAddress'] = formatted_address
+
+        user.update(valid_data)
+
+        ser_user = user_schema.dump(user)
+        return custom_json_response(ser_user, 200)
+
+
+def geocoder(streetAddress="", city="", state="", zipcode="", country=""):
+    geocoding_url = 'https://maps.googleapis.com/maps/api/geocode/json?'
+    request_url = geocoding_url + 'address=' + streetAddress + city + \
+        state + zipcode + country + '&key=' + MOSHES_GOOGLE_API_KEY
+    return requests.get(request_url)
```

Comment:
we could probably add this our helpers folder since it might be used again.

Reply:
Yeah, I also think there is google maps python API (https://github.com/googlemaps/google-maps-services-python) in case we need to do more stuff with this